includes/fbproduct.php

```diff
@@ -648,11 +650,13 @@ public function prepare_product( $retailer_id = null, $type_to_prepare_for = sel
 	 * the main function to make it easier to develop and debug, potentially
 	 * worth refactoring into main prepare_product function when complete.
 	 *
-	 * @param array $product_data map
+	 * @param array  $product_data       The preparted product data map.
+	 * @param string $google_category_id The Google product category id string.
+	 * @param string $gpc_field_name     Google product category field name.
 	 * @return array
 	 */
-	private function apply_enhanced_catalog_fields_from_attributes( $product_data, $google_category_id ) {
-		$google_category_id = $product_data['google_product_category'];
+	private function apply_enhanced_catalog_fields_from_attributes( $product_data, $google_category_id, $gpc_field_name = 'google_product_category' ) {
+		$google_category_id = $product_data[ $gpc_field_name ];
```

Comment:
This looks wrong. In the first line of the method, you overwrite a method parameter using different parameters.

Reply:
Thank you, @budzanowski, for your time and review. I did notice this redundancy (along with a few others) but decided against changing the existing logic unrelated to this PR.

I've removed the redundant logic that overrides the `$google_category_id`. 

There are changes upcoming in this method related to issue #2563.